randomwallpaper@iflow.space/ui/source_row.blp

```diff
@@ -0,0 +1,102 @@
+using Gtk 4.0;
+using Adw 1;
+
+template SourceRow : Adw.ExpanderRow {
+  title: bind source_id.text;
+  subtitle: bind enabled.active;
+
+  Box {
+    orientation: vertical;
+    spacing: 14;
+
+    Adw.EntryRow source_id {
+      title: _("Identifier - Currently only used for Generic JSON Sources");
+      input-purpose: free_form;
+      text: _("My Source - (1080p)");
+    }
+
+    Adw.ActionRow {
+      title: _("Type");
+
+      ComboBoxText source_combo {
+        valign: center;
+        active: 0;
+
+        items [
+          "Unsplash",
+          "Wallhaven",
+          "Reddit",
+          _("Generic JSON"),
+        ]
+      }
+    }
+
+    // Don't know how I can connect a change signal to a Adw.ComboRow - nothing works
```

Comment:
Sadly not. It only has `selected-id` which contains the number and `selected` which gives me the object itself.

I later ditched this aproach entirely and build the StringList dynamically from the gsettings schema enum. That has the advantage that possible values are kept at one place only.

Reply:
That is probably a good solution, will be interesting to see how you have implemented it! I experimented with reading the values from a schema for the background fill options (e.g., stretch, cover, ...) in a separate branch: https://github.com/ifl0w/RandomWallpaperGnome3/blob/b1ca7086b414cfddf6e31e057dc825d3b83d2aa7/randomwallpaper%40iflow.space/prefs.js#L142-L148

Reply:
Basically the same for the first part - but instead of a `ListStore` I used a `StringList` which can take the string array directly as input.

I'm not sure what that combox initialization is about. It looks like you've tried to render your own text? There's [`ComboBoxText`](https://docs.gtk.org/gtk4/class.ComboBoxText.html) exactly for these cases. :wink: 

> Deprecated since: 4.10
> Use [GtkDropDown](https://docs.gtk.org/gtk4/class.DropDown.html) with a [GtkStringList](https://docs.gtk.org/gtk4/class.StringList.html) instead.